src/processor/operator/aggregate/aggregate_hash_table.cpp

```diff
@@ -45,33 +47,30 @@ AggregateHashTable::AggregateHashTable(MemoryManager& memoryManager,
     initializeTmpVectors();
 }
 
-bool PartitioningAggregateHashTable::insertAggregateValueIfDistinctForGroupByKeys(
-    const std::vector<ValueVector*>& groupByFlatKeyVectors, ValueVector* aggregateVector) {
-    mergeIfFull(1);
-    return AggregateHashTable::insertAggregateValueIfDistinctForGroupByKeys(groupByFlatKeyVectors,
-        aggregateVector);
+static DataChunkState* getFirstUnflatState(const std::vector<ValueVector*>& keyVectors) {
```

Comment:
Instead of computing the first unflat state on each `AggregateHashTable::append`, I think we should consider using the leading state.